frontend/apps/reader/modules/readergesture.lua

```diff
@@ -0,0 +1,215 @@
+}
+
+function ReaderGesture:init()
+    if not Device:isTouchDevice() then return end
+    self.is_docless = self.ui == nil or self.ui.document == nil
+    self.ges_mode = self.is_docless and "gesture_fm" or "gesture_reader"
+    local gm = G_reader_settings:readSetting(self.ges_mode)
+    if gm == nil then G_reader_settings:saveSetting(self.ges_mode, {}) end
+    self.ui.menu:registerToMainMenu(self)
+    self:initGesture()
+end
+
+function ReaderGesture:initGesture()
+    local gesture_manager = G_reader_settings:readSetting(self.ges_mode)
+    -- todo: add default gesture
+    for gesture, action in pairs(gesture_manager) do
+        self:subMenuAction(gesture, action)
+    end
+end
+
+function ReaderGesture:addToMainMenu(menu_items)
+    menu_items.gesture = {
+        text = _("Gesture manager"),
+        sub_item_table = {
+            {
+                text = _("Tap right bottom corner"),
+                sub_item_table = self:tapMenu("tap_right_bottom_corner")
+            },
+            {
+                text = _("Tap left bottom corner"),
+                sub_item_table = self:tapMenu("tap_left_bottom_corner")
+            },
+            {
+                text = _("Short diagonal swipe"),
+                sub_item_table = self:tapMenu("short_diagonal_swipe")
+            },
+        },
+    }
+end
+
+function ReaderGesture:tapMenu(ges)
+    if self.is_docless then
+        return {
+            self:createSubMenu(_("Disable"), nil, ges),
+            self:createSubMenu(_("Folder up"), "folder_up", ges),
+            self:createSubMenu(_("Full screen refresh"), "full_refresh", ges),
+            self:createSubMenu(_("Night mode"), "night_mode", ges),
+            self:createSubMenu(_("Reading progress"), "reading_progress", ges),
+        }
+    else
+        return {
+            self:createSubMenu(_("Disable"), nil, ges),
+            self:createSubMenu(_("Backward 10 pages"), "page_update_down10", ges),
+            self:createSubMenu(_("Bookmarks"), "bookmarks", ges),
+            self:createSubMenu(_("Forward 10 pages"), "page_update_up10", ges),
+            self:createSubMenu(_("Full screen refresh"), "full_refresh", ges),
+            self:createSubMenu(_("Night mode"), "night_mode", ges),
+            self:createSubMenu(_("Reading progress"), "reading_progress", ges),
+            self:createSubMenu(_("Table of context"), "toc", ges),
+        }
+    end
+end
+
+function ReaderGesture:createSubMenu(text, action, ges)
+    local gesture_manager = G_reader_settings:readSetting(self.ges_mode)
+    return {
+        text = text,
+        checked_func = function()
+            return gesture_manager[ges] == action
+        end,
+        callback = function()
+            gesture_manager[ges] = action
+            G_reader_settings:saveSetting(self.ges_mode, gesture_manager)
+            self:subMenuAction(ges, action)
+        end,
+    }
+end
+
+function ReaderGesture:subMenuAction(ges, action)
```

Comment:
This function name seems wrong, and should probably be renamed to `ReaderGesture:setupGesture(ges, action)`